### ISSUE TITLE
xSchedule playlist schedule "Fire every 5 minutes" et al don't fire

### ISSUE THREAD

**Issue by merryoncherry** [smeighan/xLights#3474]
**Describe the bug**
When a playlist is scheduled to fire repeatedly, the tree view often shows the wrong time of next firing, and also these things rarely fire.

**Analysis**
The computation "GetNextFireTime" is incorrect, and also the scheduler simply isn't firing most of the time even when it knows when to fire.  I have traced these to code lines somewhat, but am far from satisfied with the quality of the patch I am running.

A reproducer is attached.  It asks for a sequence to fire every 5 minutes starting at 9:50 - 13:50

For GetNextFireTime bug (~Schedule.cpp:36):
Assume the current "now" time is 10:06 (Next scheduled time ought to be 10:10)
The "res" time gets assigned a time of now(10:06), with the minute changed to the minute of the start time(10:50).  The loops to advance this forward will have nothing to do, as it is past the current time.  10:50 will be the answer.  (In my local patch I knock the time back an hour first, works but sloppy.)

With regard to the playlist firing, all these sections of code have a conjunction similar to the following (example is from the branch for 5 minutes):
            gap.IsLongerThan(wxTimeSpan(0, 4, 59, 9999)) &&
            gap.GetMinutes() > 0 && sinceStart.IsLongerThan(wxTimeSpan(0, 4, 59, 9999)))
The 9999ms puts this at almost 5:10.  It seems this excludes triggering in most circumstances.

**To Reproduce**
Load the show folder and xSchedule file attached.
Set the schedule time to start a bit before 'now', with a larger minute value than now.

**Expected behavior**
Tree view should show the next fire time properly.
The playlist should fire when that time comes.

**Versions (please complete the following information):**
 - Windows 10
 - xSchedule 2022.19 / local build

** Other **
If you have a proposed approach for fixing this, I would be happy to work on the patch.  I can "run it", but don't know a good way to regression test it just yet.  I am currently running a patch with the milliseconds dropped to 0.  I don't know if 999 is reasonable or jittery execution could break it.

The other issue related was / is:
https://github.com/smeighan/xLights/issues/2992

**Attachments**
- Simple layout, sequence, and schedule (schedule may need modification appropriate to the time of use)
- Quick and dirty patch to show problematic code lines - sort of works but not quite properly for startup in the middle of scheduled window
[2022_xSchedule.zip](https://github.com/smeighan/xLights/files/9788011/2022_xSchedule.zip)
[xS.zip](https://github.com/smeighan/xLights/files/9788070/xS.zip)

**Comment by jholmes09** [comment]
Thanks for reporting this! I reported the original issue but was never able to go back and change the show schedule to fix. I've got new shows for 2022 that are going to use this refire technique so I hope its just a small fix.

**Comment by merryoncherry** [comment]
I've been running my patched version for a week now, seems OK.  Maybe I ought to make the PR, looks like the next days are going to be rainy and my Halloween show is finally as set up as it is going to get.

**Comment by jholmes09** [comment]
Hi @keithsw1111 I saw this thread and since I was the OP on the previous reference issue, I went to test a show and see what happens. Logs attached and show folder available here: https://iwgproductions.sharepoint.com/:u:/s/iwgprojects/EaPAPO2kNAdFgMTX59rETTkBZYmeq8hOBUL6rHHVriadMQ?e=KoSt2o

Basically, the show played correctly the first few times but after 19:30 the GUI showed next show not until 20:00 (schedule is two alternating shows that each are 'every 20 min' with a 10 minute offset on their start times. It should have been showing next show at 19:40 & 19:50, neither of which played.

Thanks!

[xSchedule_FailedFiringLog.txt](https://github.com/smeighan/xLights/files/9951125/xSchedule_FailedFiringLog.txt)
